templates/cpp-template-default/CMakeLists.txt

```diff
@@ -37,6 +37,11 @@ include(CocosBuildHelpers)
 set(BUILD_CPP_TESTS OFF CACHE BOOL "turn off build cpp-tests")
 set(BUILD_LUA_LIBS OFF CACHE BOOL "turn off build lua related targets")
 set(BUILD_JS_LIBS OFF CACHE BOOL "turn off build js related targets")
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(DEBUG_MODE ON CACHE BOOL "turn on debug mode")
+else()
+  set(DEBUG_MODE OFF CACHE BOOL "turn off debug mode")
+endif()
```

Comment:
Typically, cocos2d use option: DEBUG_MODE to control CMAKE_BUILD_TYPE, because cocos2d only test on Debug and Release mode.

Can you resolve your problem by this way?
In templates/cpp-template-default/CMakeLists.txt
```
if(DEBUG_MODE)
  set(CMAKE_BUILD_TYPE DEBUG)
else(DEBUG_MODE)
  set(CMAKE_BUILD_TYPE RELEASE)
endif(DEBUG_MODE)
```
then replace CMAKE_BUILD_TYPE with DEBUG_MODE in cmake command line.